executor/show.go

```diff
@@ -617,12 +617,45 @@ func (e *ShowExec) fetchShowCreateTable() error {
 	// add partition info here.
 	partitionInfo := tb.Meta().Partition
 	if partitionInfo != nil {
-		buf.WriteString(fmt.Sprintf("\nPARTITION BY %s ( %s ) (\n", partitionInfo.Type.String(), partitionInfo.Expr))
+		// this if statement takes care of range columns case
+		if partitionInfo.Columns != nil && partitionInfo.Type == model.PartitionTypeRange {
+			buf.WriteString(fmt.Sprintf("\nPARTITION BY RANGE COLUMNS("))
+			for i, col := range partitionInfo.Columns {
+				buf.WriteString(col.L)
+				if i < len(partitionInfo.Columns)-1 {
+					buf.WriteString(",")
+				}
+			}
+			buf.WriteString(") (\n")
+		} else {
+			buf.WriteString(fmt.Sprintf("\nPARTITION BY %s ( %s ) (\n", partitionInfo.Type.String(), partitionInfo.Expr))
+		}
 		for i, def := range partitionInfo.Definitions {
+			lessThans := ""
+			multipleLt := false
+			for j, lessThan := range def.LessThan {
+				if j > 0 {
+					lessThans = lessThans + "," + lessThan
```

Comment:
Why not use `strings.Join`?

Reply:
What is the difference between `strings.Join` and `+`?

Reply:
DONE